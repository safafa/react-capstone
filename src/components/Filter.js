/* eslint-disable react/prop-types */
import { Dropdown } from 'react-bootstrap';

const Filter = ({ handleSelect }) => {
  const letters = 'ABCDEFGHIJKLMNOPQRSTUVWYZ';
  const buttonList = letters.split('').map((letter) => <Dropdown.Item className="text-center" key={letter} eventKey={letter}>{letter}</Dropdown.Item>);
  return (
    <Dropdown onSelect={handleSelect}>
      <Dropdown.Toggle className="filter-button border-0 w-100" id="dropdown-basic">
        Filter
      </Dropdown.Toggle>
      <Dropdown.Menu className="w-100">
        {buttonList}
      </Dropdown.Menu>
    </Dropdown>
  );
};

export default Filter;
