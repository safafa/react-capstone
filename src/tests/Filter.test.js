import renderer from 'react-test-renderer';
import Filter from '../components/Filter';

describe('Filter component', () => {
  test('Snapshot test', () => {
    const tree = renderer.create(
      <Filter handleSelect={() => {}} />,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
