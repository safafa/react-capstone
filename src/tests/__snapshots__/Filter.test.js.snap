// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component Snapshot test 1`] = `
<div
  className="dropdown"
>
  <button
    aria-expanded={false}
    className="filter-button border-0 w-100 dropdown-toggle btn btn-primary"
    disabled={false}
    id="dropdown-basic"
    onClick={[Function]}
    type="button"
  >
    Filter
  </button>
</div>
`;
